### PR TITLE
Bundle Tech

### LINKED ISSUE #22: Use one tech for global scope and module system
I think it's possible to get rid of one of the techs for client and merge global and module versions into one.

// cc @mishanga

### PR DESCRIPTION
**Pull request by blond**
Resolved #22

* Merged `bh-client`, `bh-client-module` and `bh-server-include` to `bh-bundle` tech.
* Renamed `bh-server` to `bh-commonjs`.

### CODE REVIEW

File: lib/compile.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        mimic = opts.mimic || [],
+        dependencies = opts.dependencies || {},
+        depNames = Object.keys(dependencies);
+
+    // IIFE
+    file.writeLine('(function (global) {');
+
+    // Core
+    file.writeFileContent(core.path, core.contents);
+    file.writeLine('var bh = new BH();');
```
Comment (tadatuta) [inline lib/compile.js]: I think such variant is more readable:
```js
[
    'var bh = new BH();',
    'bh.setOptions({',
    '    jsAttrName: \'' + opts.jsAttrName + '\',',
    '    jsAttrScheme: \'' + opts.jsAttrScheme + '\',',
    '    jsCls: ' + (opts.jsCls ? ('\'' + opts.jsCls + '\'') : false),
    '});'
].forEach(function(line) { file.writeLine(line); });
```

File: techs/bh-bundle.js [hunk 2]
```diff
@@ -54,43 +58,38 @@ module.exports = require('enb/lib/build-flow').create()
...
-            jsCls = this._jsCls,
-            sourcemap = this._sourcemap,
-            targetPath = node.resolvePath(this._target);
+            opts = {
+                filename: node.resolvePath(this._target),
+                sourcemap: this._sourcemap,
+                jsAttrName: this._jsAttrName,
+                jsAttrScheme: this._jsAttrScheme,
+                jsCls: this._jsCls,
+                mimic: this._mimic ? [].concat(this._mimic) : [],
```
Comment (tadatuta) [inline techs/bh-bundle.js]: [].concat(this._mimic || [])

File: techs/bh-bundle.js [hunk 3]
```diff
@@ -54,43 +58,38 @@ module.exports = require('enb/lib/build-flow').create()
...
-                    data.content = bhClientProcessor.process(data.content);
-                    data.relPath = node.relativePath(file.fullname);
-                    return data;
-                });
+                return vfs.read(file.fullname, 'utf8')
+                    .then(function (content) {
+                        return {
+                            path: node.relativePath(file.fullname),
+                            contents: content
+                                .replace(/module\.exports\s*=\s*function\s*\([^\)]*\)\s*\{/, '')
```
Comment (tadatuta) [inline techs/bh-bundle.js]: would be great to add comment for regex here

File: lib/compile.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    file.writeLine('(function (global) {');
+
+    // Core
+    file.writeFileContent(core.path, core.contents);
+    file.writeLine([
+        'var bh = new BH();',
+        'bh.setOptions({',
+        '    jsAttrName: "' + opts.jsAttrName + '",',
+        '    jsAttrScheme: "' + opts.jsAttrScheme + '",',
+        '    jsCls: ' + (opts.jsCls ? ('"' + opts.jsCls + '"') : 'false'),
```
Comment (sipayRT) [inline lib/compile.js]: а зачем `false` в кавычках?

Comment (blond) [inline lib/compile.js]: При конкатенации оно всё равно превратится в строку.

**Pull request merged**